core/src/test/scala/unit/kafka/admin/ReassignPartitionsClusterTest.scala

```diff
@@ -623,6 +623,52 @@ class ReassignPartitionsClusterTest extends ZooKeeperTestHarness with Logging {
     assertEquals(Seq.empty, zkClient.getReplicasForPartition(new TopicPartition("customers", 0)))
   }
 
+  @Test
+  def shouldFinishReassignmentAfterTopicDeletion() {
+
+    //Given partitions on 3 of 3 brokers
+    val brokers = Array(100, 101, 102)
+
+    // start servers with topic deletion enabled
+    servers = brokers.map(i => createBrokerConfig(i, zkConnect, enableDeleteTopic = true, enableControlledShutdown = false, logDirCount = 3))
+      .map(c => createServer(KafkaConfig.fromProps(c)))
+
+    createTopic(zkClient, topicName, Map(
+      0 -> Seq(100, 101)
+    ), servers = servers)
+
+    val topicPartition = new TopicPartition(topicName, 0)
+    //Given a small throttle to delay partition reassignment
+    val initialThrottle = Throttle(1)
+
+    val numMessages = 1000
+    val msgSize = 100 * 1000
+    produceMessages(topicName, numMessages, acks = 0, msgSize)
+
+
+    //Start rebalance which will move replica on 100 -> replica on 102
+    val newAssignment = generateAssignment(zkClient, Array(101, 102), json(topicName), true)._1
+
+    // kick out reassignment
+    ReassignPartitionsCommand.executeAssignment(zkClient, None,
+      ReassignPartitionsCommand.formatAsReassignmentJson(newAssignment, Map.empty), initialThrottle)
+
+    //Check throttle config. Should be throttling replica 0 on 100 and 102 only.
+    checkThrottleConfigAddedToZK(adminZkClient, initialThrottle.interBrokerLimit, servers, topicName, Set("0:100","0:101"), Set("0:102"))
```

Comment:
Is this check necessary for _this_ test?
Instead, could you just `assertTrue(waitForReassignmentToComplete(0))` at the end?

Reply:
changed to waitForReassignmentToComplete().  This is a void function check for partition reassignment znode, and also use default pause-time waitForReassignmentToComplete(pause: Long = 100L)